### PR TITLE
Cleanup code regarding mimesniff 'apache bug'.

### PR DESCRIPTION
**Pull request by frewsxcv**
This change is [Reviewable](https://reviewable.io/reviews/servo/servo/10640)

### CODE REVIEW

File: components/net/mime_classifier.rs [hunk 1]
```diff
@@ -28,6 +28,20 @@ pub enum ApacheBugFlag {
     OFF
 }
 
+impl ApacheBugFlag {
+    /// https://mimesniff.spec.whatwg.org/#supplied-mime-type-detection-algorithm
+    pub fn from_content_type(last_raw_content_type: &[u8]) -> ApacheBugFlag {
+        if last_raw_content_type == b"text/plain"
+               || last_raw_content_type == b"text/plain; charset=ISO-8859-1"
+               || last_raw_content_type == b"text/plain; charset=iso-8859-1"
+               || last_raw_content_type == b"text/plain; charset=UTF-8" {
```
Comment (KiChjang) [inline components/net/mime_classifier.rs]: I'm tempted to use the mime crate for this sort of stuff.

Comment (frewsxcv) [inline components/net/mime_classifier.rs]: Maybe, though I'd imagine that should be a separate pull request? I don't know much about any mime crates

**Comment by larsbergstrom** [comment]
```
Assertion failed: (HasInsideLoopSuccs && "Loop block has no in-loop successors!"), function verifyLoop, file /Users/rustbuild/src/rust-buildbot/slave/nightly-dist-rustc-mac/build/src/llvm/include/llvm/Analysis/LoopInfoImpl.h, line 279.
Build failed, waiting for other jobs to finish...
error: Could not compile `regex`.

To learn more, run the command again with --verbose.
[Warning] Could not generate notification! Optional Python module 'pyobjc' is not installed.
Build completed in 220.61s
```

**Comment by frewsxcv** [comment]
@bors-servo retry

temporary problem?
